### PR TITLE
fix: textarea resize issue

### LINKED ISSUE #462: [FIX] Add ability to expand textarea
## Describe the bug
The textarea element cannot be extended on mobile.

## To Reproduce
1. Go to https://component.infermedica.com/?path=/story/molecules-formfield--with-textarea 
2. Try to expand the textarea using resize corner

## Expected behavior
The textarea element should be expandable on mobile.

## Screenshots
current:

expected:
Screenshot 2024-04-11 at 16 02 06

## Smartphone:
 - Device: iPhone 13 mini
 - OS: 17.4.1
 - Browser: Chrome

### PR DESCRIPTION
**Pull request by sonya0504**
## Description
The `min-height` for `textarea` is set when the component is displayed for the first time
Add `overflow-y: scroll` to enable scrolling functionality

## Related Issue

Closes #462 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/atoms/UiTextarea/UiTextarea.stories.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export const Basic: TextareaStoryType = {
   },
 };
 Basic.decorators = [ withVModel ];
+Basic.args = { resize: 'vertical' };
```
Comment (pspaczek) [inline src/components/atoms/UiTextarea/UiTextarea.stories.ts:74]: why here? Basic should use 'defaults' as initial args 🙈

File: src/components/atoms/UiTextarea/UiTextarea.vue [hunk 2]
```diff
@@ -181,8 +192,10 @@ defineExpose(
     @include mixins.use-logical($element, padding, var(--space-12) var(--space-16));
     @include mixins.use-logical($element, border, 0);
 
+    margin: var(--space-2);
     overflow: hidden;
-    max-width: 100%;
+    overflow-y: scroll;
```
Comment (pspaczek) [inline src/components/atoms/UiTextarea/UiTextarea.vue:197]:
```suggestion
    overflow-y: auto;
```
why we need scroll "for empty textarea"? 
Zrzut ekranu 2024-05-15 o 08 08 39

File: src/components/atoms/UiTextarea/UiTextarea.vue [hunk 3]
```diff
@@ -181,8 +192,10 @@ defineExpose(
     @include mixins.use-logical($element, padding, var(--space-12) var(--space-16));
     @include mixins.use-logical($element, border, 0);
 
+    margin: var(--space-2);
```
Comment (pspaczek) [inline src/components/atoms/UiTextarea/UiTextarea.vue:195]: Why 2px?

Comment (pspaczek) [reply src/components/atoms/UiTextarea/UiTextarea.vue:195]: if it's related to something maybe we should create an internal CSS var as the source of thought for both.
i.e. if it's based on border size and border-size will be changed then these values should be changed too

Comment (sonya0504) [reply src/components/atoms/UiTextarea/UiTextarea.vue:195]: Without these margins, it looks:
image

File: src/components/atoms/UiTextarea/UiTextarea.stories.ts (hunk 1 above)
Comment (pspaczek) [reply src/components/atoms/UiTextarea/UiTextarea.stories.ts:74]: ok, let we change the default `resize` in UiTextarea to `vertical` if every `textarea` use `vertical` as default :)

Comment (pspaczek) [reply src/components/atoms/UiTextarea/UiTextarea.stories.ts:74]: and remove this changes from UiTextarea stories

File: src/components/atoms/UiTextarea/UiTextarea.vue (hunk 3 above)
Comment (pspaczek) [reply src/components/atoms/UiTextarea/UiTextarea.vue:195]: Do you talk about "resize" control?

Comment (pspaczek) [reply src/components/atoms/UiTextarea/UiTextarea.vue:195]: it's related to
```
.ui-textarea {
  $this: &;
  $element: textarea;

  @include mixins.inner-border($element, $radius: var(--border-radius-form));
...
```
our border covers this control. Maybe we should use "normal" border without this hack on `after` element? Maybe it solves all the issues?

**Review by AlanLes — COMMENTED** [review]
IMO it doesn't look bad, but @pspaczek suggested that it may be solved in a simpler way. So I'll wait for you guys to check this _simpler solution_ and if it works -> great, let's proceed with it, but If not -> I'll approve this one as well :)

👀 waiting for improvement!

File: src/components/atoms/UiTextarea/UiTextarea.vue (hunk 3 above)
Comment (sonya0504) [reply src/components/atoms/UiTextarea/UiTextarea.vue]: I checked, and without this `mixin` and adding a `border` to `textarea` again the drag handle is hidden and it doesn't look like in design:
image

**Review by AlanLes — APPROVED** [review]

File: src/components/atoms/UiTextarea/UiTextarea.vue [hunk 4]
```diff
@@ -76,11 +79,12 @@ export interface TextareaEmits {
 export interface Size {
   width: string | null;
   height: string | null;
+  minHeight: string | null;
 }
 
 const props = withDefaults(defineProps<TextareaProps>(), {
   modelValue: '',
-  resize: false,
+  resize: 'vertical',
```
Comment (AlanLes) [inline src/components/atoms/UiTextarea/UiTextarea.vue:87]: It sounds like a breaking change

**Review by pspaczek — APPROVED** [review]

**Pull request merged**